service/frontend/versionChecker.go

```diff
@@ -0,0 +1,148 @@
+package frontend
+
+import (
+	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/server/api/persistenceblobs/v1"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"runtime"
+	"time"
+
+	"go.temporal.io/server/common/headers"
+	"go.temporal.io/server/common/resource"
+	"go.temporal.io/version/check"
+)
+
+const VersionCheckInterval = 24 * time.Hour
+
+type VersionChecker struct {
+	resource.Resource
+	params       *resource.BootstrapParams
+	config       *Config
+	shutdownChan chan struct{}
+	metricsScope metrics.Scope
+}
+
+func NewVersionChecker(
+	resource resource.Resource,
+	params *resource.BootstrapParams,
+	config *Config,
+) *VersionChecker {
+	return &VersionChecker{Resource: resource, params: params, config: config, shutdownChan: make(chan struct{}),
+		metricsScope: resource.GetMetricsClient().Scope(metrics.VersionCheckScope)}
+}
+
+func (vc *VersionChecker) Start() {
+	if vc.config.EnableServerVersionCheck() {
+		go vc.versionCheckLoop()
+	}
+}
+
+func (vc *VersionChecker) Stop() {
+	if vc.config.EnableServerVersionCheck() {
+		close(vc.shutdownChan)
+	}
+}
+
+func (vc *VersionChecker) versionCheckLoop() {
+	timer := time.NewTicker(VersionCheckInterval)
+	defer timer.Stop()
+	vc.performVersionCheck()
+	for {
+		select {
+		case <-vc.shutdownChan:
+			return
+		case <-timer.C:
+			vc.performVersionCheck()
+		}
+	}
+}
+
+func (vc *VersionChecker) performVersionCheck() {
+	sw := vc.metricsScope.StartTimer(metrics.VersionCheckLatency)
+	defer sw.Stop()
+	req, err := vc.createVersionCheckRequest()
+	if err != nil {
+		vc.metricsScope.IncCounter(metrics.VersionCheckFailedCount)
+		return
```

Comment:
emit a error log?

Reply:
We don't want to log intentionally. If this feature is not working due to lack of network connectivity for example we don't want to cause any concerns for our users.